npeps/npep-285-combine-crds.md

```diff
@@ -0,0 +1,100 @@
+As an API developer, I prefer to have 1 CRD for both ANP and BANP, so that I can reduce the complexity of the API and maintenance load.
+
+## API
+
+### Combining ANP and BANP
+
+To combine ANP and BANP, we need to have a way to specify policy layer. Currently, we have the following options:
+- Add a new field `spec.tier` to distinguish ANP vs BANP. The values could be `Overrride` and `Baseline`.
+- Change `spec.priority` field to have a pre-defined value for NetworkPolicy, use numbers below for ANP, and numbers above for BANP.
+For example, if we set NetworkPolicy priority to 0, then negative priorities would mean ANP, and positive priorities would mean BANP. 
+
+#### Find a better name for `spec.priority`
+
+While we are here, `priority` field name was brought to our attention multiple times as confusing. It should be mostly related to the fact
+that "higher priority" means "lower number", which is counter-intuitive. Suggested alternatives are "order", "precedence", "sequence".
+
+### Resolving differences between ANP and BANP
+
+1. BANP is a singleton, which is also the reason why it doesn't have `spec.priority` field.  
+Make BANP non-singleton and apply `spec.priority` field to it.
+
+2. ANP supports `Pass` action, while BANP doesn't. `Pass` action could be considered a delegation to the NetworkPolicy,
+  or it could be seen as return from the current policy layer.   
+We have 2 options: allow using `Pass` for BANP, meaning that the rest od BANPs will be skipped or forbid this action for BANP.  
+Unless we find strong reasons to forbid `Pass` for BANP, it makes sense to allow it to make the API more uniform.
+
+3. ANP supports `domainNames` matching for egress rules, while BANP doesn't.  
+The original reasoning from the 
+[FQDN NPEP](https://github.com/kubernetes-sigs/network-policy-api/blob/feae59d056cf87131338a8449d31b85dc1d9790f/npeps/npep-133-fqdn-egress-selector.md?plain=1#L33-L38)
+is
+> Since Kubernetes NetworkPolicy does not have a FQDN selector, adding this
+capability to BaselineAdminNetworkPolicy could result in writing baseline
+rules that can't be replicated by an overriding NetworkPolicy. For example,
+if BANP allows traffic to `example.io`, but the namespace admin installs a
+Kubernetes Network Policy, the namespace admin has no way to replicate the
+`example.io` selector using just Kubernetes Network Policies.
+
+DNS Name selector is only [supported](https://github.com/kubernetes-sigs/network-policy-api/blob/feae59d056cf87131338a8449d31b85dc1d9790f/apis/v1alpha1/adminnetworkpolicy_types.go#L282)
+with the `Allow` rule, which means the desired override by the NetworkPolicy could only be `Deny`. 
+Due to the default-deny nature of NetworkPolicy, all required traffic in a namespace should be explicitly allowed, which means
+the override of the BANP `Allow` DNS Name selector will happen automatically.
```

Comment:
I don't follow this; couldn't you have a BANP that had multiple rules:
* deny XYZ
* allow to foo.example.com
* deny to ABC
* allow to bar.example.com

Reply:
Sure, you can. This part tries to address the original concern that a NetworkPolicy can't override `allow to foo.example.com` part of BANP, which it can, because an override will have to be "deny". So even if we ever implement `allow DNSName` peer in network policy, it has nothing to do with the ability to override BANP.